app/api/data-fetching/[...slug]/route.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  HttpForbiddenError,
+  HttpNotFound,
+  HttpUnauthorizedError,
+} from '#clients/exceptions';
+import { Exception } from '#models/exceptions';
+import { hasRights } from '#models/user/rights';
+import { ISession } from '#models/user/session';
+import logErrorInSentry, { logInfoInSentry } from '#utils/sentry';
+import { withAntiScrapping } from '#utils/server-side-helper/app/with-anti-bot';
+import { APIHandler, APIPath, APIRoutesScope } from '../routes';
+
+async function getRoute(
+  _request: Request,
+  { params }: { params: { slug: Array<string> } },
+  /* The `session` parameter in the `getRoute` function is an object of type `ISession` representing
+  the user session. It contains information about the current user's session, such as user details,
+  permissions, and any other relevant data needed to process the request. The `session` object is
+  used to check if the user has the necessary rights or scopes to access a specific API route before
+  handling the request. */
+  session: ISession
+) {
+  const slug = params.slug.at(-1) as string;
+  const route = params.slug.slice(0, -1).join('/');
```

Comment:
This would be nice in a helper as it is done twice in the file

I would maybe add sentry logs as it looks bug prone